packages/web-components/src/components/cbp-toggle/cbp-toggle.tsx

```diff
@@ -39,18 +39,46 @@ export class CbpToggle {
     });
   }
 
+  @Listen('keydown')
+  handleKeyDown( ev: KeyboardEvent){
+    if(ev.key === ' '){
+      ev.preventDefault(); //techdebt: needed to prevent scrolling when using this keypress
+      this.toggleEvent();
+    }
+  }
+
+  
+  @Listen('change')
+  handleChange() {
+    this.toggleEvent();
+  }
+
+  @Event() toggleClick: EventEmitter;
+
   /** Event: toggles the control true/false & updates DOM accordingly*/
   toggleEvent(){
-    //TODO: logic here
+    let checkbox = this.host.querySelector('input[type="checkbox"]');
```

Comment:
This should be a component-scoped variable that you get on componentDidLoad. The disabled prop needs to push the down the attribute on the native input. See checkbox for an example. (this will set you up for implementing the ID part later as well).

It looks like I assign the event listener that way as well (rather then using @Listen) and invoke the handler directly. Checkbox also emits a custom event, which toggle should also do to be consistent.

Reply:
updated as discussed